cfme/services/requests.py

```diff
@@ -1,286 +1,370 @@
+        # GCE
+        availability_zone = SummaryFormItem('Placement - Options', 'Availability Zones')
+        cloud_network = SummaryFormItem('Placement - Options', 'Cloud Network')
+        # Infra
+        datacenter = SummaryFormItem('Datacenter', 'Name')
+        cluster = SummaryFormItem('Cluster', 'Name')
+        resource_pool = SummaryFormItem('Resource Pool', 'Name')
+        folder = SummaryFormItem('Folder', 'Name')
+        host_filter = SummaryFormItem('Host', 'Filter')
+        host_name = SummaryFormItem('Host', 'Name')
+        datastore_storage_profile = SummaryFormItem('Datastore', 'Storage Profile')
+        datastore_filter = SummaryFormItem('Datastore', 'Filter')
+        datastore_name = SummaryFormItem('Datastore', 'Name')
+
+    @View.nested
+    class hardware(Tab):  # noqa
+        num_cpus = SummaryFormItem('Hardware', 'Number of CPUS')
+        memory = SummaryFormItem('Hardware', 'Startup Memory (MB)')
+        dynamic_memory = SummaryFormItem('Hardware', 'Dynamic Memory')
+        vm_limit_cpu = SummaryFormItem('VM Limits', 'CPU (%)')
+        vm_reserve_cpu = SummaryFormItem('VM Reservations', 'CPU (%)')
+
+    @View.nested
+    class network(Tab):  # noqa
+        vlan = SummaryFormItem('Network Adapter Information', 'vLan')
+
+    @View.nested
+    class properties(Tab):  # noqa
+        instance_type = SummaryFormItem('Properties', 'Instance Type')
+        boot_disk_size = SummaryFormItem('Properties', 'Boot Disk Size ')
+        is_preemtible = Checkbox(name='hardware__is_preemptible')
+
+    @View.nested
+    class customize(Tab):  # noqa
+        username = SummaryFormItem('Credentials', 'Username')
+        ip_mode = SummaryFormItem('IP Address Information', 'Address Mode')
+        hostname = SummaryFormItem('IP Address Information', 'Address Mode')
+        subnet_mask = SummaryFormItem('IP Address Information', 'Subnet Mask')
+        gateway = SummaryFormItem('IP Address Information', 'Gateway')
+        dns_server_list = SummaryFormItem('DNS', 'DNS Server list')
+        dns_suffix_list = SummaryFormItem('DNS', 'DNS Suffix list')
+        subnet_mask = SummaryFormItem('IP Address Information', 'Subnet Mask')
+        customize_template = SummaryFormItem('Customize Template', 'Script Name')
+
+    @View.nested
+    class schedule(Tab):  # noqa
+        when_provision = SummaryFormItem('Schedule Info', 'When to Provision')
+        stateless = Checkbox(name='shedule__stateless')
+        power_on = SummaryFormItem('Lifespan', 'Power on virtual machines after creation')
+        retirement = SummaryFormItem('Lifespan', 'Time until Retirement')
+        retirement_warning = SummaryFormItem('Lifespan', 'Retirement Warning')
+
+    breadcrumb = BreadCrumb()
+    toolbar = RequestDetailsToolBar()
+    reason = Input(name='reason')
+    submit = Button(title='Submit')
+    cancel = Button(title="Cancel this provisioning request")
+
+    def approve(self, reason_text, cancel=False):
```

Comment:
I  suspect these checks and calls have to be moved to request methods. You cannot check location, etc inside view.